scss/_material-checkbox.scss

```diff
@@ -23,92 +23,82 @@ $md-checkmark-color: $white;
   }
 
   label {
-    margin-bottom: 0;
     padding-left: $md-checkbox-size + 5px;
+    margin-bottom: 0;
   }
 
-  i.md-checkbox-control {
+  .md-checkbox-control {
     cursor: pointer;
 
-    &:before,
-    &:after {
-      content: "";
+    &::before,
+    &::after {
       position: absolute;
-      left: 0;
       top: 0;
+      left: 0;
+      content: "";
     }
 
-    &:before {
+    &::before {
       // box
       width: $md-checkbox-size;
       height: $md-checkbox-size;
+      cursor: pointer;
       background: #fff;
       border: 2px solid $md-checkbox-border-color;
-      border-radius: 2px;
-      cursor: pointer;
-      transition: background .3s;
-    }
-
-    &:after {
-      // checkmark
+      @include border-radius(2px);
+      @include transition(background .3s);
     }
   }
 
-  input[type="checkbox"] {
-    outline: 0;
+  [type="checkbox"] {
     display: none;
+    outline: 0;
 
-    &:checked,
-    &.indeterminate {
-      + i.md-checkbox-control {
-        &:before {
-          background: $md-checkbox-checked-color;
-          border: none;
-        }
-        &:after {
-          $md-checkmark-size: $md-checkbox-size - 2*$md-checkbox-padding;
-
-          transform: rotate(-45deg);
-
-          top: ($md-checkbox-size / 2) - ($md-checkmark-size / 4) - $md-checkbox-size/10;
-          left: $md-checkbox-padding;
-          width: $md-checkmark-size;
-          height: $md-checkmark-size / 2;
+    &:disabled {
+      + .md-checkbox-control {
```

Comment:
Again here why drop the `i`

Reply:
https://stylelint.io/user-guide/rules/selector-no-qualifying-type